File: GuessNumberGame.py
import random
import sys

def main():
    number_to_guess = random.randint(1, 100)
    number_of_tries = 0
    has_guessed_correctly = False

    print("欢迎来到猜数字游戏！")
    print("我已经想好了一个1到100之间的数字。")

    while not has_guessed_correctly:
        try:
            guess = int(input("请输入你的猜测: "))
        except ValueError:
            print("请输入一个有效的数字。")
            continue
        number_of_tries += 1

        if guess < number_to_guess:
            print("太小了，再试一次！")
        elif guess > number_to_guess:
            print("太大了，再试一次！")
        else:
            has_guessed_correctly = True
            print(f"恭喜你，猜对了！你总共猜了 {number_of_tries} 次。")

    # 将猜测次数保存到文件
    try:
        with open("guess_results.txt", "a") as writer:
            writer.write(f"用户猜对数字 {number_to_guess} 共尝试了 {number_of_tries} 次。\n")
    except IOError as e:
        print(f"保存文件时发生错误: {e.strerror}")

if __name__ == "__main__":
    main()

File: GuessNumberGame.java
我import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;
import java.util.Scanner;

public class GuessNumberGame {
    public static void main(String[] args) {
        Random random = new Random();
        int numberToGuess = random.nextInt(100) + 1;
        Scanner scanner = new Scanner(System.in);
        int numberOfTries = 0;
        boolean hasGuessedCorrectly = false;

        System.out.println("欢迎来到猜数字游戏！");
        System.out.println("我已经想好了一个1到100之间的数字。");

        while (!hasGuessedCorrectly) {
            System.out.print("请输入你的猜测: ");
            int guess = scanner.nextInt();
            numberOfTries++;

            if (guess < numberToGuess) {
                System.out.println("太小了，再试一次！");
            } else if (guess > numberToGuess) {
                System.out.println("太大了，再试一次！");
            } else {
                hasGuessedCorrectly = true;
                System.out.println("恭喜你，猜对了！你总共猜了 " + numberOfTries + " 次。");
            }
        }

        scanner.close();

        // 将猜测次数保存到文件
        try (FileWriter writer = new FileWriter("guess_results.txt", true)) {
            writer.write("用户猜对数字 " + numberToGuess + " 共尝试了 " + numberOfTries + " 次。\n");
        } catch (IOException e) {
            System.out.println("保存文件时发生错误: " + e.getMessage());
        }
    }
}